### PR TITLE
Add `rb-escape_utils`

### PR DESCRIPTION
**Pull request by svengiegerich**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/rb-escape_utils/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+source:
+  - url: https://rubygems.org/downloads/{{ name }}-{{ version }}.gem
+    sha256: dffb7010922880ace6ceed642156c64e2a64620f27e0849f43bc4f68fd3c2c09
+
+build:
+  number: 0
+  script:
+    - gem install -N -l -V --norc --ignore-dependencies {{ name }}-{{ version }}.gem
+    - gem unpack {{ name }}-{{ version }}.gem
+    - make -C {{ PREFIX }}/lib/ruby/gems/{{ ruby }}.0/gems/{{ name }}-{{ version }}/ext/{{ name }} clean
```
Comment (janjagusch) [inline recipes/rb-escape_utils/meta.yaml:17]: @svengiegerich, you call `make` here. where did you get these build instructions from and what happens when you remove this line?

File: recipes/rb-escape_utils/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (janjagusch) [inline recipes/rb-escape_utils/LICENSE.txt:1]: TODO: We should check if the unpacked gem (`gem unpack {{ name }}-{{ version }}.gem`) contains the license.

Comment (janjagusch) [reply recipes/rb-escape_utils/LICENSE.txt:1]: The gem file actually contains the LICENSE:

```
gem unpack ~/Downloads/escape_utils-1.3.0.gem
Unpacked gem: '/Users/janjagusch/escape_utils-1.3.0'

ls escape_utils-1.3.0/LICENSE
escape_utils-1.3.0/LICENSE
```

So we should be able to make a slight adjustment in the recipe and then don't have to put the LICENSE manually into the feedstock.

Comment (janjagusch) [reply recipes/rb-escape_utils/LICENSE.txt:1]: @svengiegerich, you can delete this file now.

File: recipes/rb-escape_utils/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  number: 0
+  script:
+    - gem install -N -l -V --norc --ignore-dependencies {{ name }}-{{ version }}.gem
+    - gem unpack {{ name }}-{{ version }}.gem
+    - cp {{ name }}-*/LICENSE .
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (janjagusch) [inline recipes/rb-escape_utils/meta.yaml:22]:
```suggestion
    - {{ compiler('c') }}
    - make
```

File: recipes/rb-escape_utils/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: Faster string escaping routines for your ruby apps
+  doc_url: https://www.rubydoc.info/gems/{{ name }}
+  dev_url: https://github.com/brianmario/escape_utils
+
+extra:
+  recipe-maintainers:
+    - svengiegerich
```
Comment (janjagusch) [inline recipes/rb-escape_utils/meta.yaml:44]: happy to co-maintain this. :)

```suggestion
    - svengiegerich
    - janjagusch
```

**Review by janjagusch — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**